presto-main/src/test/java/com/facebook/presto/sql/TestExpressionInterpreter.java

```diff
@@ -584,6 +586,42 @@ public void testCastToString()
         // TODO enabled when DECIMAL is default for literal: assertOptimizedEquals("cast(12345678901234567890.123 as VARCHAR)", "'12345678901234567890.123'");
     }
 
+    @Test
+    public void testCastBigintToBoundedVarchar()
+    {
+        assertEvaluatedEquals("CAST(12300000000 AS varchar(11))", "'12300000000'");
+        assertEvaluatedEquals("CAST(12300000000 AS varchar(50))", "'12300000000'");
+
+        try {
+            evaluate("CAST(12300000000 AS varchar(3))", true);
+            fail("Expected to throw an INVALID_CAST_ARGUMENT exception");
+        }
+        catch (PrestoException e) {
+            try {
+                assertEquals(e.getErrorCode(), INVALID_CAST_ARGUMENT.toErrorCode());
+                assertEquals(e.getMessage(), "Value 12300000000 cannot be represented as varchar(3)");
+            }
+            catch (Throwable failure) {
+                failure.addSuppressed(e);
+                throw failure;
+            }
+        }
+
+        try {
+            evaluate("CAST(-12300000000 AS varchar(3))", true);
+        }
+        catch (PrestoException e) {
```

Comment:
nit: Could this testing logic be put into a private static "assertThrows" (or something like that) helper method so as to match the style of the rest of this class? It could be helpful in the future too to test exceptions being thrown properly, and would be less awkward than catching an exception thrown by an assertion.

Reply:
I will refactor it in PR #17232